src/movies.js

```diff
@@ -1,22 +1,72 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(moviesArray) {
+    let directors = moviesArray.map(function (movie){
+      return movie.director;
```

Comment:
EXTRA EXERCISE: try to simplify your code -> convert into arrow function /could be one line ;) You can apply this to all iterations.